### PR TITLE
Lps 145190 reviewed

### PR DESCRIPTION
**Pull request by arthurchan35**
Hey @stian-sigvartsen 
I have reorganized your commits a little bit, but changes are still under your commit info.
The reason I did is that I noticed some of your original commits have mixed change content under a commit(e.g. a test commit containing SF of other class, etc).

Now the commits are logically grouped.

Also see https://github.com/liferay-appsec/liferay-portal/pull/553/commits/c6c293c1569a9a2f59061d3442cf00cad619e2c1#diff-18eaf6593dd7e95c35400f3d640b418b1673ad68808c27342d49ce3b00d87804R433 for audience validation.

### CODE REVIEW

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayJWTBearerGrantHandler.java [hunk 1]
```diff
@@ -162,8 +174,16 @@ protected boolean isGrantHandlerEnabled() {
 		return _oAuth2ProviderConfiguration.allowJWTBearerGrant();
 	}
 
+	private <U, V> void _addDefaults(Map<U, V> map, Map<U, V> defaultsMap) {
+		if (defaultsMap != null) {
+			defaultsMap.forEach(map::putIfAbsent);
+		}
+
+		_rebuild(CompanyConstants.SYSTEM);
```
Comment (arthurchan35) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayJWTBearerGrantHandler.java:182]: I cannot bring myself to believe that we need this line, but I am also not confident enough to make a commit to remove it. So I am leaving a comment here. @stian-sigvartsen

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:security

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:security

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:relevant

**Review by stian-sigvartsen — COMMENTED** [review]
The commits make much more sense now, thanks @arthurchan35 .
The new audience validation works fine for a single virtualhost, but if there are multiple, then there is a race condition between setting the audience and validating with it. I added a simple fix which can probably be optimized.

Your integration test was great for the happy path. I just integrated with the other "client.test" package and used the existing base class there to reduce duplication. Also this enables us to test combination flows easier in the future.

I added an audience validation test.

Please see liferay-appsec/liferay-portal/pull/564

**Pull request closed without merging**